src/views/clusteroverview/SettingsTab/TemplatesProjectTab/utils/utils.ts

```diff
@@ -0,0 +1,60 @@
+import HyperConvergedModel from '@kubevirt-ui/kubevirt-api/console/models/HyperConvergedModel';
+import { TemplateModel, V1Template } from '@kubevirt-utils/models';
+import { k8sDelete, k8sGet, k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+
+import { HyperConverged, TemplateList } from './../../../utils/types';
+
+const TYPE_LABEL = 'template.kubevirt.io/type';
+const BASE = 'base';
+
+export const OPENSHIFT = 'openshift';
+
+export const getCurrentTemplatesNamespaceFromHCO = (hyperConverged: HyperConverged): string =>
+  hyperConverged?.spec?.commonTemplatesNamespace || OPENSHIFT;
+
+export const updateHCOCommonTemplatesNamespace = async (
+  hyperConverged: HyperConverged,
+  newNamespace: number | string | null,
+  handelError: (value: string) => void,
+  handleLoading: (value: boolean) => void,
+) => {
+  const currentTemplatesNamespace = getCurrentTemplatesNamespaceFromHCO(hyperConverged);
+  if (newNamespace !== currentTemplatesNamespace) {
+    handleLoading(true);
+    try {
+      await k8sPatch<HyperConverged>({
+        model: HyperConvergedModel,
+        resource: hyperConverged,
+        data: [
+          {
+            op: 'replace',
+            path: `/spec/commonTemplatesNamespace`,
+            value: newNamespace === OPENSHIFT ? null : newNamespace,
```

Comment:
why do you need to check here if the newNamespace is openshift? what will happen if we changed it once from `openshift` to some other namespace and than I want to change it back? won't we lose all templates?